smart_open/webhdfs.py

```diff
@@ -114,11 +114,11 @@ def read(self, size=None):
 
         try:
             while len(self._buf) < size:
-                self._buf += self._response.raw.readline()
+                self._buf += self._response.raw.read(io.DEFAULT_BUFFER_SIZE)
         except StopIteration:
             pass
 
-        self._buf, retval = self._buf[size:], retval[:size]
+        self._buf, retval = self._buf[size:], self._buf[:size]
```

Comment:
Wow. What was this doing before?

Reply:
I think it worked in the base case only (read all content from beginning to end).